[Bug]: storyStoreV7: false breaks build

### Describe the bug

Build failing when `features.storyStoreV7` is set to `false` in `main.ts`.

```
resolve 'C:/SB v7/node_modules/@storybook/angular/dist/client/config.js-generated-config-entry.js' in 'C:\SB v7'
  using description file: C:\SB v7\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: C:\SB v7\node_modules\@storybook\angular\package.json (relative path: ./dist/client/config.js-generated-config-entry.js)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js doesn't exist
      .mjs
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.mjs doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.js doesn't exist
      .jsx
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.jsx doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.ts doesn't exist
      .tsx
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.tsx doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.json doesn't exist
      .cjs
        Field 'browser' doesn't contain a valid alias configuration
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js.cjs doesn't exist
      as directory
        C:\SB v7\node_modules\@storybook\angular\dist\client\config.js-generated-config-entry.js doesn't exist
c:\SB v7\node_modules\webpack\lib\Compilation.js:2016
                                        const notFoundError = new ModuleNotFoundError(
                                                              ^

ModuleNotFoundError: Module not found: Error: Can't resolve 'C:/SB v7/node_modules/@storybook/angular/dist/client/docs/config-generated-config-entry.js' in 'C:\SB v7
```


### To Reproduce

Download and run this stackblitz project: https://stackblitz.com/edit/github-hr4cka-q4mnyi

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
  Binaries:
    Node: 16.14.2 - C:\Program Files\nodejs\node.EXE
    npm: 8.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 111.0.5563.65
    Edge: Spartan (44.19041.1266.0), Chromium (111.0.1661.44)
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-rc.4 => 7.0.0-rc.4
    @storybook/addon-interactions: ^7.0.0-rc.4 => 7.0.0-rc.4
    @storybook/addon-links: ^7.0.0-rc.4 => 7.0.0-rc.4
    @storybook/angular: ^7.0.0-rc.4 => 7.0.0-rc.4
    @storybook/blocks: ^7.0.0-rc.4 => 7.0.0-rc.4
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1
```


### Additional context

_No response_

--- a/code/frameworks/angular/src/server/framework-preset-angular-docs.ts
+++ b/code/frameworks/angular/src/server/framework-preset-angular-docs.ts
@@ -1,8 +1,7 @@
-import path from 'path';
 import { StorybookConfig } from '@storybook/types';
 import { hasDocsOrControls } from '@storybook/docs-tools';
 
 export const previewAnnotations: StorybookConfig['previewAnnotations'] = (entry = [], options) => {
   if (!hasDocsOrControls(options)) return entry;
-  return [...entry, path.join(__dirname, '../../dist/client/docs/config')];
+  return [...entry, require.resolve('../client/docs/config')];
 };

--- a/code/lib/builder-webpack5/src/preview/iframe-webpack.config.ts
+++ b/code/lib/builder-webpack5/src/preview/iframe-webpack.config.ts
@@ -1,4 +1,4 @@
-import { dirname, join, resolve } from 'path';
+import { dirname, isAbsolute, join, resolve } from 'path';
 import { DefinePlugin, HotModuleReplacementPlugin, ProgressPlugin, ProvidePlugin } from 'webpack';
 import type { Configuration } from 'webpack';
 import HtmlWebpackPlugin from 'html-webpack-plugin';
@@ -112,6 +112,12 @@ export default async (
         if (typeof entry === 'object') {
           return entry.absolute;
         }
+
+        // TODO: Remove as soon as we drop support for disabled StoryStoreV7
+        if (isAbsolute(entry)) {
+          return entry;
+        }
+
         return slash(entry);
       }
     ),

